build_tools/github_actions/repo-diff.py

```diff
@@ -0,0 +1,1231 @@
+"""
+TheRock Repository Diff Report Generator
+
+OVERVIEW:
+This script generates comprehensive HTML reports highlighting commit differences for each component
+between two TheRock builds. It analyzes submodule changes, superrepo component changes, and provides
+visual indicators for different types of changes (added, removed, changed, unchanged, reverted).
+
+HOW IT WORKS:
+1. Fetches submodule information from two TheRock commits using GitHub API
+2. Compares submodules to identify added, removed, changed, unchanged, and reverted components
+3. For superrepos (rocm-libraries, rocm-systems), analyzes component-level changes within projects/ and shared/ directories
+4. For regular submodules, fetches commit history between the two versions
+5. Detects reverted submodules by comparing commit timestamps
+6. Generates styled HTML report with visual indicators and commit details
+7. Creates GitHub Actions step summary for CI integration
+
+USAGE IN CI:
+- Automatically runs in ci_nightly.yml workflow to compare latest build with last successful build
+- Uses --find-last-successful flag to automatically find the previous successful run
+- Generates reports for nightly build comparisons and failure analysis
```

Comment:
This level of detail seems more appropriate for the PR description or a markdown file, not the script itself (which should be general enough that it could be used from multiple different workflows)